onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -856,10 +857,15 @@ class PlannerImpl {
                     // we have seen
                     plan_.SetLocation(static_cast<size_t>(index), exec_provider->GetOrtDeviceByMemType(OrtMemType::OrtMemTypeDefault));
                   } else {
-                    // Default the location to CPU
-                    plan_.SetLocation(static_cast<size_t>(index),
-                                      execution_providers_.Get(CPU)->GetOrtDeviceByMemType(OrtMemType::OrtMemTypeDefault));
-                    set_implicitly_consumed_node_arg_has_heterogenous_ep_consumers.insert(index);
+                    if (utils::DoesEpLocationRequiresAdjustment(exec_provider->Type())) {
```

Comment:
`DoesEpLocationRequiresAdjustment` feels very allocation planner specific. Does that need to be in the general utils or could it be a local function here? 

Would 'UseEpSpecificCpuAllocator' be clearer? I _think_ that would be the high level description of what you're doing, and that setting a location is a side effect of that.